public/pages/CreateMonitor/components/AnomalyDetectors/EmptyFeaturesMessage/EmptyFeaturesMessage.js

```diff
@@ -6,48 +6,145 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { EuiEmptyPrompt, EuiButton, EuiText, EuiLoadingChart } from '@elastic/eui';
-import { OPENSEARCH_DASHBOARDS_AD_PLUGIN } from '../../../../../utils/constants';
-
-const EmptyFeaturesMessage = (props) => (
-  <div
-    style={{
-      borderRadius: '5px',
-      padding: '10px',
-      border: '1px solid #D9D9D9',
-      height: '250px',
-      width: '100%',
-      ...props.containerStyle,
-    }}
-  >
-    {props.isLoading ? (
-      <EuiEmptyPrompt style={{ maxWidth: '45em' }} body={<EuiLoadingChart size="xl" />} />
-    ) : (
-      <EuiEmptyPrompt
-        style={{ maxWidth: '45em' }}
-        body={
-          <EuiText>
-            No features have been added to this anomaly detector. A feature is a metric that is used
-            for anomaly detection. A detector can discover anomalies across one or more features.
-          </EuiText>
-        }
-        actions={[
-          <EuiButton
-            data-test-subj="createButton"
-            href={`${OPENSEARCH_DASHBOARDS_AD_PLUGIN}#/detectors/${props.detectorId}/features`}
-            target="_blank"
-          >
-            Add Feature
-          </EuiButton>,
-        ]}
-      />
-    )}
-  </div>
-);
+import { OPENSEARCH_DASHBOARDS_AD_PLUGIN, AD_PREVIEW_DAYS } from '../../../../../utils/constants';
+
+const ErrorType = {
+  PREVIEW_EXCEPTION: 0,
+  NO_FEATURE: 1,
+  NO_ENABLED_FEATURES: 2,
+  SPARSE_DATA: 3,
+};
+
+/**
+ *
+ * @param {string} errorType error type
+ * @param {*} err error message
+ * @param {*} isHCDetector whether the detector is HC
+ * @returns error messages to show on the empty trigger page
+ */
+function getErrorMsg(errorType, err, isHCDetector) {
+  switch (errorType) {
+    case ErrorType.PREVIEW_EXCEPTION:
+      return err;
+    case ErrorType.NO_FEATURE:
+      return 'No features have been added to this anomaly detector. A feature is a metric that is used for anomaly detection. A detector can discover anomalies across one or more features.';
+    case ErrorType.NO_ENABLED_FEATURES:
+      return 'No features have been enabled in this anomaly detector. A feature is a metric that is used for anomaly detection. A detector can discover anomalies across one or more features.';
+    case ErrorType.SPARSE_DATA:
+      return `No sample anomaly result generated. Please check detector interval and make sure you have >400 data points${
+        isHCDetector ? ' for some entities ' : ' '
+      }in the last ${AD_PREVIEW_DAYS} days.`;
+    default:
+      console.log('unexpected error type: ' + errorType);
+      return '';
```

Comment:
will this just return empty error message back to customers? Should we add something there to avoid this situation?

Reply:
I added the empty string check in the consumer side in EmptyFeatureMessage.js. Will only show error when it is a non-empty string.

```
function getErrorType(err, features) {
  // if error is a non-empty string, return it.
  if (err) return ErrorType.PREVIEW_EXCEPTION;

  if (features === undefined || features.length == 0) {
    return ErrorType.NO_FEATURE;
  }
```